plugins/com.google.cloud.tools.eclipse.appengine.libraries.test/src/com/google/cloud/tools/eclipse/appengine/libraries/Bug3380Test.java

```diff
@@ -0,0 +1,149 @@
+    factory.setNamespaceAware(true);
+  }
+
+  private Pom pom;
+  private IFile pomFile;
+  private final XPath xpath = xpathFactory.newXPath();
+
+  @Before
+  public void setUp() throws SAXException, IOException, CoreException {
+    xpath.setNamespaceContext(new MappedNamespaceContext("m", "http://maven.apache.org/POM/4.0.0"));
+
+    IProject project = projectCreator.getProject();
+    pomFile = project.getFile("pom.xml");
+    try (InputStream in =
+        Files.newInputStream(Paths.get("testdata/testBug3380.xml").toAbsolutePath())) {
+      pomFile.create(in, IFile.FORCE, null);
+      pom = Pom.parse(pomFile);
+    }
+
+    Logger logger = Logger.getLogger(ArtifactRetriever.class.getName());
+    logger.setLevel(Level.OFF);
+  }
+
+  @After
+  public void tearDown() {
+    Logger logger = Logger.getLogger(ArtifactRetriever.class.getName());
+    logger.setLevel(null);
+  }
+
+  @Test
+  public void testBug3380() throws CoreException {
+    LibraryFile addFile = new LibraryFile(coordinates("com.google.cloud", "google-cloud-asset"));
+    Library addLib = newLibrary("asset", addFile);
+
+    LibraryFile existFile = new LibraryFile(coordinates("com.google.cloud", "google-cloud-automl"));
+    Library existLib = newLibrary("automl", existFile);
+
+    LibraryFile removeFile =
+        new LibraryFile(coordinates("com.google.cloud", "google-cloud-bigquery"));
+    Library removeLib = newLibrary("bigquery", removeFile);
+
+    pom.updateDependencies(Arrays.asList(addLib, existLib), Arrays.asList(removeLib));
+
+    Assert.assertTrue(pom.dependencyManaged("com.google.cloud", "google-cloud-asset"));
+    Assert.assertTrue(pom.dependencyManaged("com.google.cloud", "google-cloud-automl"));
+
+    Element bomElement = null;
+    try {
+      bomElement = (Element) xpath.evaluate(
+          "//m:dependencyManagement/m:dependencies/m:dependency[m:groupId='com.google.cloud'][m:artifactId='google-cloud-bom']",
+          pom.document.getDocumentElement(), XPathConstants.NODE);
+    } catch (XPathExpressionException e) {
+      fail(e.getMessage());
+    }
+    Assert.assertNotNull(bomElement); // Make sure BOM element added
+
+    Element dependencies = null;
+    try {
+      dependencies = (Element) xpath.evaluate("./m:dependencies", pom.document.getDocumentElement(),
+          XPathConstants.NODE);
+    } catch (XPathExpressionException e) {
+      fail(e.getMessage());
+    }
+    Assert.assertNotNull(dependencies);
+
+    Element dependency = pom.findDependency(dependencies, "com.google.cloud", "google-cloud-asset");
+    Assert.assertNotNull(dependency);
+
+    Node versionElement = Pom.findChildByName(dependency, "version");
+    Assert.assertNull(versionElement); // Version should be removed If BOM added
+
+    dependency = pom.findDependency(dependencies, "com.google.cloud", "google-cloud-automl");
+    Assert.assertNotNull(dependency);
+
+    versionElement = Pom.findChildByName(dependency, "version");
```

Comment:
I don't think you need to do this a second time.

Reply:
There are two different dependencies. One is a new one and another one already exists.